docs/rfcs/038-independent-compute-release.md

```diff
@@ -0,0 +1,298 @@
+
+TL;DR it has several important attributes:
+
+- Revision -- unique release ID/primary key. Not the same as application version,
+  because the same version can be deployed several times, e.g. after newer version rollback.
+- App version -- version of the application chart / code.
+- Config -- set of overrides to the default config of the application.
+- Status -- current status of the release in the history.
+- Timestamps -- tracks when release was created and deployed.
+
+## Proposed implementation
+
+### Separate release branch
+
+We will use a separate release branch `release-compute` to have a clean history fo releases and commits.
+In order to avoid confusion with storage releases, we will use a different prefix for compute [git release
+tags](https://github.com/neondatabase/neon/releases) -- `release-compute-XXXX`. It probably still makes
+sense to use just `release-XXXX` for docker images, because `neondatabase/compute-node-vXX` is a separate
+Docker image from `neondatabase/neon` and `neondatabase/compute-node-v16:release-compute-XXXX` will just
+look awkward. Yet, it doesn't matter much, so we could do either of two.
+
+### Compute config/settings manifest
+
+We will create a new sub-directory `compute` and file `compute/manifest.yaml` with a structure:
+
+```yaml
+pg_settings:
+  # Common settings for primaries and secondaries of all versions.
+  common:
+    wal_log_hints: "off"
+    max_wal_size: "1024"
+
+  per_version:
+    14:
+      # Common settings for both replica and primary of version PG 14
+      common:
+        shared_preload_libraries: "neon,pg_stat_statements,extension_x"
+    15:
+      common:
+        shared_preload_libraries: "neon,pg_stat_statements,extension_x"
+      # Settings that should be applied only to
+      replica:
+        # Available only starting Postgres 15th
+        recovery_prefetch: "off"
+    # ...
+    17:
+      common:
+        # For example, if third-party `extension_x` is not yet available for PG 17
+        shared_preload_libraries: "neon,pg_stat_statements"
+      replica:
+        recovery_prefetch: "off"
+```
+
+N.B.: setting value should be a string with `on|off` for booleans and a number (as a string)
+without units for all numeric settings. That's how control plane currently operates.
+
+The priority of settings will be (higher number is a higher priority):
+
+1. Any static and hard-coded settings in the control plane
+2. `pg_settings->common`
+3. Per-version `common`
+4. Per-version `replica`
+5. Any per-user/project/endpoint overrides in the control plane
+6. Any dynamic setting calculated based on the compute size
+
+Compute team will own the content of `compute/manifest.yaml`.
+
+### Control plane: releases table
+
+In order to store information about releases, control plane will use a table `compute_releases` with the following
+schema:
+
+```sql
+CREATE TABLE compute_releases (
+  -- Unique release ID
+  revision BIGSERIAL PRIMARY KEY,
```

Comment:
It's worth noting that the `revision` for a compute tag will be different on each region, since it's assigned automatically by the cplane db. That's OK, but might be confusing. I guess we have that with helm revisions already.